tests/utils/dockercli/volumelifecycle.go

```diff
@@ -85,7 +90,21 @@ func ReadFromVolume(ip, volName, containerName, fileName string) (string, error)
 // DeleteVolume helper deletes the created volume as per passed volume name.
 func DeleteVolume(ip, name string) (string, error) {
 	log.Printf("Destroying volume [%s]\n", name)
-	return ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
+	var out string
+	var err error
+	for attempt := 0; attempt <= maxAttempt; attempt++ {
+		out, err = ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
+		if err != nil {
+			if strings.Contains(out, "Plugin initialization in progress.") {
+				misc.SleepForSec(waitTime)
+				log.Printf("Volume cannot be deleted yet as plugin initialization still in progress. Retrying...")
+				continue
+			}
+			return out, err
+		}
+		break
+	}
+	return out, err
```

Comment:
Can avoid nesting.
You only need to wait when error is not nil (exit status 1) and output message contains the error string

```
	var out string
	var err error

	for attempt := 0; attempt <= maxAttempt; attempt++ {
		out, err := ssh.InvokeCommand(ip, dockercli.RemoveVolume+name)
		if err != nil && strings.Contains(out, "Plugin initialization in progress.") {
			misc.SleepForSec(waitTime)
			log.Printf("Volume cannot be deleted yet as plugin initialization still in progress. Retrying...")
			continue
		} else {
			break
		}
	}

	return out, err
```